bouncer/src/repo/core/model/bson/repo_bson_factory.cpp

```diff
@@ -25,6 +25,203 @@
 
 using namespace repo::core::model;
 
+/**
+* Specialisation of the BSON Builder dedicated to building Meshes and
+* Supermeshes. This is for internal use by the RepoBSONFactory and is
+* intended to build immutable nodes via one of the makeMeshNode
+* overloads.
+*/
+class RepoMeshBSONBuilder : public RepoBSONBuilder
```

Comment:
Is there any reason why it's hidden in here?

I understand it's not used elsewhere, but there's probably no harm in separating it out since it's a sub class of RepoBSONBuilder it would make more sense to live with the RepoBSONBuilder

or it's own file (probably prefer that)